File: Solutions/1572. Matrix Diagonal Sum/1572. Matrix Diagonal Sum.py
class Solution:
    def diagonalSum(self, mat: List[List[int]]) -> int:
        n,res=len(mat),0
        for i in range(n):
            res+=mat[i][i]+mat[i][n-i-1]
        return res-mat[n//2][n//2] if n%2==1 else res

File: Solutions/1572. Matrix Diagonal Sum/1572. Matrix Diagonal Sum.go
func diagonalSum(mat [][]int) int {
    n,sum:=len(mat),0
    for i,_:=range(mat){
        sum+=mat[i][i]+mat[i][n-i-1]
    }
    if n%2==1{
        return sum-mat[n/2][n/2]
    }
    return sum
}